admin/app.js

```diff
@@ -32,7 +32,34 @@ const logger = require('./services/log.service').getLogger()
 const sqlService = require('./services/data-access/sql.service')
 const app = express()
 setupLogging(app)
-sqlService.initPool()
+
+/**
+ * Sleep in ms
+ * @param ms - milliseconds
+ * @return {Promise}
+ */
+function sleep (ms) {
+  return new Promise(resolve => {
+    setTimeout(resolve, ms)
+  })
+}
+
+// Initialise the Database Connection pool
+;(async function () {
+  try {
+    logger.debug('Attempting to initialise the connection pool')
+    await sqlService.initPool()
+  } catch (error) {
+    logger.alert('Failed to connect to the database', error)
+    // The initial probe connection was not able to connection: the DB is not available.  This will cause all
+    // connections in the connection pool to be initialised to closed connections. By pausing, we allow time for the
+    // db to become available.  When run in a docker container the PM2 process manager will restart the process, and
+    // hopefully the DB will be up by then.
+    await sqlService.drainPool()
+    await sleep(30000)
+    process.exit(1)
```

Comment:
Isn't the exit part covered by drainPool?

Reply:
@GeorgeChatzigiannis not sure where you get this idea from?  drain pool is for draining a pool, not for exiting a process.  these are 2 completely separate tasks.  i raised a similar concern on your recent PR

Reply:
@GeorgeChatzigiannis here is the code...
```
sqlService.drainPool = async () => {
  if (!pool) {
    logger.warn('The connection pool is not initialised')
    return
  }
  return pool.close()
}
```

Reply:
@GeorgeChatzigiannis - what @GuyHarwood said.

Reply:
👍